### PR TITLE
feat: notify hummock snapshot to frontend in meta

### PR DESCRIPTION
**Pull request by HuaHuaY**
## What's changed and what's your intention?

I added to meta the ability to push hummock snapshots to the frontend. Frontend only needs to call `pin_snapshot` for an `epoch` when the epoch is used for the first time. And meta will send notification to frontend every time `commit_epoch` is called.

I add a `HummockSnapshotManager` in frontend as a cache of hummock snapshot.
I modified the meaning of `pin_snapshot`. Now `pin_snapshot` have an argument `epoch` instead of `last_pinned`, which is the snapshot that we want to pin. Meta receives the rpc request. Then meta will compare `epoch` and `max_commited_epoch`. 
1. If `epoch` is less than or equal to `max_commited_epoch` and `epoch` was never pinned, it will pin `max_commited_epoch` and return `max_commited_epoch`.
2. If `epoch` is less than or equal to `max_commited_epoch` and `epoch` was pinned, it will return `epoch`.
3. If `epoch` is bigger than `max_commited_epoch`, it will return error.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Comment by HuaHuaY** [comment]
Sorry for the long wait for new commits.

**Comment by hzxa21** [comment]
LGTM

cc @zwang28 PTAL. After this PR, the pin_snapshot behavior has changed to: unpin all pinned epoch that is within (last_pinned, max_committed_epoch) and pin max committed epoch if not pin.

**Review by zwang28 — APPROVED** [review]
LGTM for the rest.

File: src/meta/src/hummock/hummock_manager.rs [hunk 1]
```diff
@@ -864,6 +869,14 @@ where
             self.check_state_consistency().await;
         }
 
+        self.env
+            .notification_manager()
+            .notify_frontend(
```
Comment (zwang28) [inline src/meta/src/hummock/hummock_manager.rs:874]: Seems notify_frontend can be blocking due to `NotificationManagerCore::notify_frontend`?
`HummockManager::commit_epoch` shouldn't be blocked by it.

**Pull request merged**